### PR TITLE
c2c: deflake and refactor TestTenantReplicationProtectedTimestampManagement

### LINKED ISSUE #106473: ccl/streamingccl/streamingest: TestTenantReplicationProtectedTimestampManagement failed
ccl/streamingccl/streamingest.TestTenantReplicationProtectedTimestampManagement [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10830894?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10830894?buildTab=artifacts#/) on release-23.1 @ [5a53114bf5ae9e5ec672d43acf599f145bf22341](https://github.com/cockroachdb/cockroach/commits/5a53114bf5ae9e5ec672d43acf599f145bf22341):

```
=== RUN   TestTenantReplicationProtectedTimestampManagement
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/90c1b75d835f45b8488807abb5b1092d/logTestTenantReplicationProtectedTimestampManagement728671638
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestTenantReplicationProtectedTimestampManagement
    replication_stream_e2e_test.go:897: -- test log scope end --
--- FAIL: TestTenantReplicationProtectedTimestampManagement (335.05s)
=== RUN   TestTenantReplicationProtectedTimestampManagement/pause-before-terminal=false
    --- FAIL: TestTenantReplicationProtectedTimestampManagement/pause-before-terminal=false (160.87s)
=== RUN   TestTenantReplicationProtectedTimestampManagement/pause-before-terminal=false/complete-replication=true
    testutils.go:180: error scanning '&{0xc025031320 <nil>}': pq: cutover time 1688800035.591457000,0 is before earliest safe cutover time 1688800036.173821419,0
        --- FAIL: TestTenantReplicationProtectedTimestampManagement/pause-before-terminal=false/complete-replication=true (79.12s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantReplicationProtectedTimestampManagement.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Jira issue: CRDB-29577

### PR DESCRIPTION
**Pull request by msbutler**
In #106473, the test harness attempted to cutover to a timestamp that was older than the replication job's retained timestamp. This doesn't seem that suprising since the ttl in the test is 1 second.

To prevent future flakes, the harness now cuts over to the latest replicated time.

Fixes #106473

Release note: None

### CODE REVIEW

File: pkg/ccl/streamingccl/replicationtestutils/testutils.go [hunk 1]
```diff
@@ -171,16 +171,23 @@ func (c *TenantStreamingClusters) WaitUntilStartTimeReached(ingestionJobID jobsp
...
 func (c *TenantStreamingClusters) Cutover(
 	producerJobID, ingestionJobID int, cutoverTime time.Time, async bool,
 ) {
 	// Cut over the ingestion job and the job will stop eventually.
 	var cutoverStr string
-	c.DestSysSQL.QueryRow(c.T, `ALTER TENANT $1 COMPLETE REPLICATION TO SYSTEM TIME $2::string`,
-		c.Args.DestTenantName, cutoverTime).Scan(&cutoverStr)
-	cutoverOutput := DecimalTimeToHLC(c.T, cutoverStr)
-	require.Equal(c.T, cutoverTime, cutoverOutput.GoTime())
+	if cutoverTime.Equal(time.Time{}) {
```
Comment (stevendanna) [inline pkg/ccl/streamingccl/replicationtestutils/testutils.go:181]: If you fancy: `cutoverTime.IsZero()`?

**Review by stevendanna — APPROVED** [review]
Thanks!

File: pkg/ccl/streamingccl/replicationtestutils/testutils.go (hunk 1 above)
Comment (msbutler) [reply pkg/ccl/streamingccl/replicationtestutils/testutils.go]: done.

**Comment by msbutler** [comment]
in the update, I also shortened the te test name to TestProtectedTimestampManagement.

**Comment by msbutler** [comment]
TFTR!

bors r=stevendanna

**Pull request merged**